staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -226,9 +243,16 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 
 			start := strings.LastIndex(js[:syntax.Offset], "\n") + 1
 			line := strings.Count(js[:start], "\n")
-			return fmt.Errorf("json: line %d: %s", line, syntax.Error())
+			return JSONSyntaxError{
+				Line: line,
+				Err:  fmt.Errorf(syntax.Error()),
+			}
 		}
 	}
+
+	if err != nil && err != io.EOF && hasPrefix([]byte(err.Error()), []byte("yaml: line")) {
```

Comment:
No, upon looking at the function that is eventually called in the `ghodss/yaml` package from the `Decode` method implemented in our `pkg/util/yaml/decoder.go`, errors prefixed with yaml only occur when a call to [gopkg.in/yaml.v2#Unmarshal](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/ghodss/yaml/yaml.go#L82) results in an error. The only case that returns an error without the `yaml:` prefix is if [ghodss/yaml/yaml.go#convertToJSONableObject](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/ghodss/yaml/yaml.go#L89) fails, which will only happen [if an unknown key type is given](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/ghodss/yaml/yaml.go#L164)